project/ScalazBuild.scala

```diff
@@ -4,13 +4,12 @@ import Keys._
 object Scalaz extends Build {
   val testDeps = Seq("org.scalacheck" %% "scalacheck" % "1.13.4" % "test")
 
-  def module(prjName: String) = Project(
-    id = prjName,
-    base = file(prjName)).settings(
+  def stdSettings(prjName: String) = Seq(
     name := s"scalaz-$prjName",
-    scalaVersion := "2.11.11",
+    scalaVersion := "2.12.2",
     scalacOptions ++= Seq("-feature","-deprecation", "-Xlint", "-language:higherKinds",
-                          "-Ydelambdafy:method", "-Ypartial-unification", "-target:jvm-1.8"),
+                          "-Ydelambdafy:method", "-Ypartial-unification", "-target:jvm-1.8", "-opt:l:project",
```

Comment:
It would interesting to have opinion of @retronym on such thing, as he knows how such a library is crafted and he is probably well aware of the current state of the inliner.

I don't feel qualified enough to make a decision here, I'm just pushing on the safe side due to bad memories about inlining... But it would be a shame to miss such improvement if they are now properly implemented... might be worth trying initially... and can go back later...

Thanks Kenji for the pointer!